integration/query_test.go

```diff
@@ -835,7 +835,7 @@ func TestQueryCommandLimitPushDown(t *testing.T) {
 
 		len            int                 // expected length of results
 		queryPushdown  resultPushdown      // optional, defaults to noPushdown
-		limitPushdown  bool                // optional, set true for expected pushdown for limit
+		limitPushdown  resultPushdown      // optional, set true for expected pushdown for limit
```

Comment:
`TestQueryCommandLimitPushDown` test currently tests against `shareddata.Composites`, we would want to test with `shareddata.Scalars` provider for testing filters that are boolean, string and so on like below.

```
filter: bson.D{{"v", true}},
```

```
filter: bson.D{{"v", bson.D{{"$eq", true}}}},
```

```
filter: bson.D{{"v", bson.D{{"$ne", true}}}},
```

```
filter: bson.D{{"v", bson.D{{"$gt", true}}}},
```

Reply:
`shareddata.Scalars` provider and a lot more tests are added. Please see if there are still some missing cases.